packages/js-drive/lib/core/getRandomQuorumFactory.js

```diff
@@ -0,0 +1,107 @@
+const BufferWriter = require('@dashevo/dashcore-lib/lib/encoding/bufferwriter');
+const Hash = require('@dashevo/dashcore-lib/lib/crypto/hash');
+const QuorumsNotFoundError = require('./errors/QuorumsNotFoundError');
+const ValidatorSet = require('../validator/ValidatorSet');
+const llmqMap = require('./llmqMap');
+
+const MIN_QUORUM_VALID_MEMBERS = 90;
+
+/**
+ * Calculates scores for validator quorum selection
+ * it calculates sha256(hash, modifier) per quorumHash
+ * Please note that this is not a double-sha256 but a single-sha256
+ *
+ * @param {Buffer[]} quorumHashes
+ * @param {Buffer} modifier
+ * @return {Object[]} scores
+ */
+function calculateQuorumHashScores(quorumHashes, modifier) {
+  return quorumHashes.map((hash) => {
+    const bufferWriter = new BufferWriter();
+
+    bufferWriter.write(hash);
+    bufferWriter.write(modifier);
+
+    return { score: Hash.sha256(bufferWriter.toBuffer()), hash };
+  });
+}
+
+/**
+ *
+ * @param {RpcClient} coreRpcClient
+ * @return {getRandomQuorum}
+ */
+function getRandomQuorumFactory(coreRpcClient) {
+  /**
+   * Gets the current validator set quorum hash for a particular core height
+   *
+   * @typedef {getRandomQuorum}
+   * @param {SimplifiedMNList} sml
+   * @param {number} quorumType
+   * @param {Buffer} entropy - the entropy to select the quorum
+   * @param {number} coreHeight
+   * @return return {Promise<QuorumEntry>} - the current validator set's quorumHash
+   */
+  async function getRandomQuorum(sml, quorumType, entropy, coreHeight) {
+    const validatorQuorums = sml.getQuorumsOfType(quorumType);
+
+    if (validatorQuorums.length === 0) {
+      throw new QuorumsNotFoundError(sml, quorumType);
+    }
+
+    const allValidatorQuorumsExtendedInfo = await coreRpcClient.quorum('listextended');
```

Comment:
I've just realized that it's not gonna work when you reply old blocks. You need to get quorum information based on the current core chain locked height.

Reply:
Ody is gonna add height to RPC when he has time.